addons/l10n_fr_account/models/account_move.py

```diff
@@ -20,3 +20,17 @@ def _get_view(self, view_id=None, view_type='form', **options):
     def _compute_l10n_fr_is_company_french(self):
         for record in self:
             record.l10n_fr_is_company_french = record.country_code in record.company_id._get_france_country_codes()
+
+    @api.depends('country_code', 'move_type')
+    def _compute_show_delivery_date(self):
+        # EXTENDS 'account'
+        super()._compute_show_delivery_date()
+        for move in self:
+            if move.country_code == 'FR':
+                move.show_delivery_date = move.is_sale_document()
+
+    def _post(self, soft=True):
+        for move in self:
+            if move.country_code == 'FR' and move.is_sale_document() and not move.delivery_date:
+                move.delivery_date = move.invoice_date or fields.Date.context_today(self)
+        return super()._post(soft)
```

Comment:
Care when doing a super at the end of the function you could have a case where the delivery date is modified after in the exection. So might be better to put it at the beginning of the function right ? :smile: 
Also when doing an extension always put the Extend comment :smile: